src/update-function/swap-array-control.ts

```diff
@@ -0,0 +1,24 @@
+import { SwapArrayControlAction } from '../actions';
+import { formArrayReducer } from '../array/reducer';
+import { FormArrayState, isArrayState } from '../state';
+import { ensureState } from './util';
+
+/**
+ * This update function takes a source index, a destination index and returns a projection function
+ * that swaps the child controls at the source and destination indices in a form array state.
+ */
+export function swapArrayControl(fromIndex: number, toIndex: number): <TValue>(state: FormArrayState<TValue>) => FormArrayState<TValue>;
```

Comment:
This is a really minor thing, but for the swap operation I feel the comment should not talk about source and destination indices and the parameters should also not be `from` and `to`. Instead for the comment I would use

    This update function takes two indices and returns a projection function that swaps the child controls at those indices in a form array state.

The parameters could be called `index1`, and `index2`.

For the uncurried variant it's the same:

    This update function takes a form array state and two indices and swaps the child controls at those indices in the state.